lib/services/access_request_test.go

```diff
@@ -51,6 +51,125 @@ func (s *AccessRequestSuite) TestRequestMarshaling(c *C) {
 	}
 }
 
+// TestPluginDataExpectations verifies the correct behavior of the `Expect` mapping.
+func (s *AccessRequestSuite) TestPluginDataExpectations(c *C) {
+	const rname = "my-resource"
+	const pname = "my-plugin"
+	data, err := NewPluginData(rname, KindAccessRequest)
+	c.Assert(err, IsNil)
+
+	// Set two keys, expecting them to be unset.
+	err = data.Update(PluginDataUpdateParams{
+		Kind:     KindAccessRequest,
+		Resource: rname,
+		Plugin:   pname,
+		Set: map[string]string{
+			"hello": "world",
+			"spam":  "eggs",
+		},
+		Expect: map[string]string{
+			"hello": "",
+			"spam":  "",
+		},
+	})
+	c.Assert(err, IsNil)
```

Comment:
Should return `trace.CompareFailed` and use `fixtures.ExpectCompareFailed` to make sure the right error is returned